### DISCUSSION TITLE
Set CA certificate start- and end-date

### DISCUSSION THREAD

**Discussion by JayFoxRox** [mitmproxy/mitmproxy#6505] [Ideas]
I have an IoT device and by firmware analysis I seem to have stumbled upon a "feature" that lets it skip cert verification if the cert is expired (it just ignores the cert then and allows traffic, also mentioning this in the logs).
Namely, it normally checks for `/CN=<important-to-get-right>/`.

My IoT device is using a different firmware revision (undumped) so I'm not sure if this "feature" works on my device.
However, I don't know how to work with certificates (meaning I don't know how to generate such a CA cert) and I'd prefer if mitmproxy gave me the option to provide an expired certificate.

I'd be grateful if someone could show me how to generate a new mitmproxy CA cert with a specified date range.
Ideally this would already be a feature of mitmweb (more control over CA cert).

**Comment by mhils** [comment]
https://docs.mitmproxy.org/stable/concepts-certificates/ has some instructions on certificate generations. Additions/updates are welcome!

For your particular use case, you can maybe also just delete your `~/.mitmproxy` folder, set back your system clock the desired amount of time, and then start mitmproxy? 😄

**Reply by JayFoxRox** [reply]
Yes I did attempt this by modifying https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/certs.py#L28-L29
The certificate appears to be expired then, but looks like my device doesn't care.

I might have a different bug elsewhere though. I still think this would be a handy feature.
The current certificates already have shortened lifespan because of some changes in Chrome/Firefox. When sniffing other devices, there might be more checks about expected certificate expiration or different code paths for expired certificates (which might be beneficial for mitm-attacks).
